### PR TITLE
[Area Chart] Support legend multi selection

### PR DESCRIPTION
**Pull request by srmukher**
Support legend multi selection for area chart

image

### CODE REVIEW

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 1]
```diff
@@ -870,8 +869,14 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
 
-  private _getCircleRadius = (xDataPoint: number, circleRadius: number, circleId: string): number => {
-    const { isCircleClicked, nearestCircleToHighlight, activePoint } = this.state;
+  private _getCircleRadius = (xDataPoint: number, circleRadius: number, circleId: string, legend: string): number => {
+    const { isCircleClicked, nearestCircleToHighlight, activePoint, selectedLegends } = this.state;
+
+    // Show the circle if no legends are selected or if the point's legend is in the selected legends
+    if (selectedLegends.length > 0 && !selectedLegends.includes(legend)) {
+      return 0;
+    }
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:878]: this is missing out on legend hovered condition.

File: packages/charts/react-charting/etc/react-charting.api.md [hunk 2]
```diff
@@ -194,6 +194,7 @@ export interface IAccessibilityProps {
```
Comment (fabricteam) [inline packages/charts/react-charting/etc/react-charting.api.md:194]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33475&commitId=0cb14835e191964bc2ae2e90ff8bda4f570ff659&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Left center.chromium.png|2616| Changed|
|Callout.Bottom Right Edge RTL.chromium.png|1593| Changed|
|Callout.Top center.chromium.png|2178| Changed|
|Callout.Bottom auto edge.chromium.png|2309| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/etc/react-charting.api.md [hunk 3]
```diff
@@ -194,6 +194,7 @@ export interface IAccessibilityProps {
 
 // @public
 export interface IAreaChartProps extends ICartesianChartProps {
+    canSelectMultipleLegends?: boolean;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/etc/react-charting.api.md:197]: >canSelectMultipleLegends [](http://example.com/codeflow?start=4&length=24)

remove

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 4]
```diff
@@ -621,10 +612,16 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
         enabledWrapLines={this.props.enabledLegendsWrapLines}
         focusZonePropsInHoverCard={this.props.focusZonePropsForLegendsInHoverCard}
         {...this.props.legendProps}
+        canSelectMultipleLegends={this.props.canSelectMultipleLegends}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:615]: >canSelectMultipleLegends={this.props.canSelectMultipleLegends} [](http://example.com/codeflow?start=8&length=62)

this is not needed

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 5]
```diff
@@ -621,10 +612,16 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
+        onChange={this._onLegendChange}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:616]: >_onLegendChange [](http://example.com/codeflow?start=23&length=15)

onLegendsChange

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 6]
```diff
@@ -930,14 +936,18 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
     const found: any = this._calloutPoints.find((e: { x: string | number }) => e.x === modifiedXVal);
     // Show details in the callout for the focused point only
     found.values = found.values.filter((e: { y: number }) => e.y === y);
+    const filteredValues =
+      this.state.selectedLegends.length > 0
+        ? found.values.filter((value: { legend: string }) => this.state.selectedLegends.includes(value.legend))
+        : found.values;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:942]: move logic to function

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 1 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:878]: updated to use _legendHighlighted

File: packages/charts/react-charting/etc/react-charting.api.md (hunk 3 above)
Comment (srmukher) [reply packages/charts/react-charting/etc/react-charting.api.md:197]: done

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 5 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:616]: renamed to _onLegendSelectionChange

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 6 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx:942]: moved

File: packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 4 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/AreaChart/AreaChart.base.tsx]: removed

**Comment by AtishayMsft** [comment]
CI build is failing.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**